settings/provisioner/openstack/network/default.yml

```diff
@@ -0,0 +1,40 @@
+---
+# This is the most common network topology for OpenStack Testing deployments
+#
+provisioner:
+
+    networks:
+        data:
+            name: '{{ run.uuid }}-data'
+        external:
+            name: '{{ run.uuid }}-external'
+        management: 
+            name: '{{ run.uuid }}-management'
+
+    subnets:
+        data:
+            name: !lookup provisioner.networks.data.name
+            network_name: !lookup provisioner.networks.data.name
+            allocation_pool_start: 172.16.1.10
+            allocation_pool_start: 172.16.1.100
+            cidr: 172.16.1.0/24
+        external:
+            name: !lookup provisioner.networks.external.name
+            network_name: !lookup provisioner.networks.external.name
+            allocation_pool_start: 172.31.0.10
+            allocation_pool_start: 172.31.0.100
+            cidr: 172.31.0.0/23
+        management:
+            name: !lookup provisioner.networks.management.name
+            network_name: !lookup provisioner.networks.management.name
+            allocation_pool_start: 192.168.1.10
+            allocation_pool_start: 192.168.1.100
+            cidr: 192.168.1.0/24
+
+    rotuers:
+        router:
+            name: '{{ run.uuid }}-router'
+            attach_subnets:
+                - !lookup provisioner.subnets.external.name
```

Comment:
Do not reference here to another lookup. You can just do !lookup provisioner.networks.external.name
The same is actual for the second item in the list 

Also I think all the lookup should be surrounded with "{{ }}"